cloudflare.sh

```diff
@@ -63,30 +65,30 @@ test "$1" = "verify" && {
 
 idsfile=$(mktemp /tmp/cloudflare-ids-XXXXXX)
 zonefile=$(mktemp /tmp/cloudflare-zone-XXXX)
-cat > $zonefile
-trap "rm -f $idsfile $zonefile $zonefile-data" EXIT
+cat > "$zonefile"
+trap 'rm -f $idsfile $zonefile $zonefile-data' EXIT
 
 DOMAIN_ID=$(apicmd GET /zones | jq -r '.result | .[] | .name + ":" + .id' \
-            | grep $DOMAIN) \
+            | grep "$DOMAIN") \
   || exit 1
-DOMAIN_ID=$(echo $DOMAIN_ID | cut -d: -f2)
+DOMAIN_ID=$(echo "$DOMAIN_ID" | cut -d: -f2)
 
 # only edit existing TXT records containig the "v=spf1" marker
 apicmd GET "/zones/$DOMAIN_ID/dns_records?type=TXT" \
-  | jq -r '.result | .[] | select ( .content | contains("v=spf1") ) | .name + ":" + .id' > $idsfile
+  | jq -r '.result | .[] | select ( .content | contains("v=spf1") ) | .name + ":" + .id' > "$idsfile"
 
-while read line
+while read -r line
 do
-  name=$(echo $line | cut -d" " -f1)
-  content=$(echo $line | cut -d^ -f2 | tr -d \")
-  id_to_change=$(grep "^$name" $idsfile | cut -d: -f2)
+  name=$(echo "$line" | cut -d" " -f1)
+  content=$(echo "$line" | cut -d^ -f2 | tr -d \")
+  id_to_change=$(grep "^$name" "$idsfile" | cut -d: -f2)
 
-  echo -n "Changing $name with id $id_to_change... "
-  cat > $zonefile-data <<EOF
+  printf "%s" "Changing $name with id $id_to_change... "
```

Comment:
since we are using printf, wouldn't be better to use `printf "Changing %s with id %s... " "$name" "$id_to_change"` ?

Reply:
Yes. I appreciate your feed-back. Thanks!